source/common/router/router.cc

```diff
@@ -147,14 +148,28 @@ void Filter::chargeUpstreamCode(const Http::HeaderMap& response_headers,
 
       Http::CodeUtility::chargeResponseStat(info);
     }
+
+    if (upstream_host) {
+      const uint64_t response_code = Http::Utility::getResponseStatus(info.response_headers_);
+      if (dropped) {
+        upstream_host->stats().rq_dropped_.inc();
+      } else if (response_code == 200) {
```

Comment:
If we are going by normal HTTP semantics, this should be `!Http::CodeUtility::is5xx` IMO